Add compatibility macros for `dbt-audit-helper`

morsapaes opened the pull request:
## Motivation

Fixes #19. Will push a new tag once it's approved!

## Tips for reviewer

Verified that the integrations tests run successfully for `dbt-audit-helper`.

benesch commented on README.md:
```diff
@@ -26,15 +27,17 @@ packages:
     version: 0.8.0
   - package: MaterializeInc/materialize_dbt_utils
-    version: 0.3.0
+    version: 0.5.0
```
I think you wanted v0.4.0, yeah?

```suggestion
    version: 0.4.0
```

morsapaes replied:
Duh.

benesch commented on CHANGELOG.md:
```diff
@@ -10,7 +14,7 @@
 * Correctly mark the `insert_by_period` macro as unsupported.
 
-## 0.2.0 - 2021-02-14
+## 0.2.0 - 2022-02-14
```
🙈 thanks!

benesch commented on README.md:
```diff
@@ -4,6 +4,7 @@ This [dbt](https://github.com/dtb-labs/dbt) package provides shims for using the
 following packages with [Materialize]:
 
 - [dbt-utils](https://github.com/dbt-labs/dbt-utils)
+- [dbt-audit-helper](https://github.com/fishtown-analytics/dbt-audit-helper)
```
```suggestion
- [dbt-audit-helper](https://github.com/dbt-labs/dbt-audit-helper)
```

benesch commented on README.md:
```diff
@@ -26,15 +27,17 @@ packages:
   - macro_namespace: dbt_utils
     search_order: [materialize_dbt_utils, dbt_utils]
+  - macro_namespace: audit_helper
+        search_order: ['materialize_dbt_utils', 'audit_helper']
```
```suggestion
  - macro_namespace: audit_helper
    search_order: [materialize_dbt_utils, audit_helper]
```

benesch commented on integration_tests/dbt_audit_helper/dbt_project.yml:
```diff
@@ -0,0 +1,37 @@
+name: materialize_dbt_utils_dbt_utils_integration_tests
```
```suggestion
name: materialize_dbt_utils_dbt_audit_helper_integration_tests
```

benesch commented on integration_tests/dbt_audit_helper/dbt_project.yml:
```diff
@@ -0,0 +1,37 @@
+  - macro_namespace: audit_helper
+    search_order: [materialize_dbt_utils, audit_helper]
+
+seeds:
+  dbt_utils_integration_tests:
+    schema_tests:
+      # Materialize does not support the `lead` window function.
+      data_test_mutually_exclusive_ranges_no_gaps:
+        +enabled: false
+      data_test_mutually_exclusive_ranges_with_gaps:
+        +enabled: false
+      data_test_mutually_exclusive_ranges_with_gaps_zero_length:
+        +enabled: false
+
+      # Materialize does not support the `lag` window function.
+      data_test_sequential_timestamps:
+        +enabled: false
+      data_test_sequential_values:
+        +enabled: false
+
+models:
+  dbt_utils_integration_tests:
+    datetime:
+      # Tested by test_date_spine_override.
+      test_date_spine:
+        +enabled: false
+    schema_tests:
+      # Tested by test_recency_override.
+      test_recency:
+        +enabled: false
```
Not positive, but I think all this stuff doesn't apply to audit helper!

```suggestion
```

morsapaes replied:
Ah! Need to give it another pass to double check there aren't more copy+paste leftovers!

morsapaes commented on macros/dbt_audit_helper/compare_column_values.sql:
```diff
@@ -0,0 +1,60 @@
+select
+    match_status,
+    count_records,
+    round(100.0 * count_records / sum(count_records) over (), 2) as percent_of_total
```
Ups, missed this one in the commit.

benesch reviewed (commented):
Neat! Line comments within, then some overall questions:

1. Can you update CI to run these tests too? https://github.com/MaterializeInc/materialize-dbt-utils/blob/e5577f6a6faca03389bacdbe43a009cbfdefcb65/.github/workflows/main.yml#L36-L38
2. In theory, we could instead create a new MaterializeInc/materialize-dbt-audit repository for these shims. Does dbt have guidance on which option to choose? It seems like [tsql-utils](https://github.com/dbt-msft/tsql-utils#compatibility) goes this route, so that seems like pretty solid evidence that we're doing it right, but I wanted to double check.

morsapaes commented:
> Can you update CI to run these tests too?

Ah, thanks! I missed that.

> In theory, we could instead create a new MaterializeInc/materialize-dbt-audit repository for these shims. Does dbt have guidance on which option to choose?

I based this on the `tsql-utils` example, which sounds a little saner than having a separate repo for each package we might want to shim. I'll double check with Anders, though!

morsapaes commented:
Brought up the question of where to put things in the [dbt community Slack](https://getdbt.slack.com/archives/C030A0UF5LM/p1651763770291259), and so far it seems like there isn't a best practice yet:

> yeah this is a Question. one we’ve been thinking on a lot. it’s right at the intersection of adapters and packages. dbt Labs (and I) have some ideas, but I’m not sure if anything’s official yet. I want to say there’s a notion currently gaining consensus that all the shim macros in dbt-utils move into the adapter repos?

, but some projects (like Firebolt) have started maintaining shim macros within the adapter repo.

benesch reviewed (approved):
LGTM! Keeping it all in one repo sounds good, since it sounds like the question on the table is whether to merge repositories even _further_ and combine this repository with the dbt adapter istelf.

Note that I didn't actually look at the implementation of any of these macros, as I assume they're all just copied from the PostgreSQL implementations!

morsapaes commented:
Thanks, @benesch! The macros are mostly copied from the Postgres implementation, with the exception of a slight twist in `compare_column_values.sql` and `compare_queries.sql` to work around an unsupported (window) function that I replaced with a harmless hack (if such a thing exists 👹). Added a note to both so we know it's there, though!
